lib/to_do/handlers/to_do.ex

```diff
@@ -17,4 +17,10 @@ defmodule ToDo.Handlers.ToDo do
       _ -> value
     end
   end
+
+  def handle(%Request{method: "DELETE", full_path: [_path, id | _]} = _req) do
+    [key | _] = id |> String.split("/")
+    API.delete(key)
```

Comment:
`key` is an implementation detail.  Shouldn't the API just take the `id`, and let whatever underlying implementation figure out its a key.  When a function knows about its underlying implementation details that's a potential Law of Demeter violation.

Reply:
I've moved the parsing into API.delete:
```
  def delete(path) do
    id = parse_id_from_path(path)

    DB.all()
    |> DB.delete_todo(id)
    |> DB.save()
  end

  defp parse_id_from_path(path) do
    [id | _] = path |> String.split("/")
    id
  end
```

Let me know if this isn't what you had in mind!

Reply:
I hope that explanation is a bit clearer than the first.

Reply:
it also appears that the feedback here is tied to the [other comment](https://github.com/kelseyroy/http_server/pull/34#discussion_r1269756945)

Reply:
This was helpful! I just talked to Taylor about this too and I was definitely trying to over optimize (which is the point you were making in the other comment as well). I've gone ahead and pushed up a commit that adds `:id` on Request and `parse_id_from_path/1` to Request.Parser!